src/batch/src/executor/join/hash_join_state.rs

```diff
@@ -232,7 +232,9 @@ impl<K: HashKey> ProbeTable<K> {
     pub(super) fn set_probe_data(&mut self, probe_data_chunk: DataChunk) -> Result<()> {
         self.build_data_chunk()?;
         let probe_data_chunk = probe_data_chunk.compact()?;
-        ensure!(probe_data_chunk.cardinality() > 0);
+        // TODO(yuhao): We should make sure the output chunk of upstream executor
+        // has cardinality > 0.
+        // ensure!(probe_data_chunk.cardinality() > 0);
```

Comment:
Why remove it here?

Reply:
Because we can't ensure this here until we check all executor implementations to make sure output chunk has cardinality > 0